### PR TITLE
Complex pow

### PR DESCRIPTION
**Pull request by Beakerboy**

### CODE REVIEW

File: src/Number/Complex.php [hunk 1]
```diff
@@ -377,6 +377,35 @@ public function divide($c): Complex
         }
     }
 
+    /**
+     * Complex exponentiation
+     * Raise a complex number to a power.
+     * https://en.wikipedia.org/wiki/Complex_number#Multiplication_and_division
```
Comment (markrogoyski) [inline src/Number/Complex.php:383]: Copy and paste mistake?
Maybe link here:
- https://en.wikipedia.org/wiki/Complex_number#Exponentiation
- https://mathworld.wolfram.com/ComplexExponentiation.html

**Comment by markrogoyski** [comment]
Thanks for this pull request. Looks good. I'll fix the link.

**Comment by markrogoyski** [comment]
Maybe I was too quick to merge it.

This test case is producing a complex with `NAN + NANi`.
```php
            [
                ['r' => 0, 'i' => 0],
                ['r' => 0, 'i' => 0],
                ['r' => 1, 'i' => 0],
            ],
```

R thinks it is this:
```R
> complex(real=0, imaginary=0)**complex(real=0, imaginary=0)
[1] 1+0i
```
And Python:
```python
In [115]: complex(0,0)**complex(0,0)
Out[115]: (1+0j)
```

**Comment by Beakerboy** [comment]
> In algebra and combinatorics, the generally agreed upon value is 0^0 = 1, whereas in mathematical analysis, the expression is sometimes left undefined. Computer programming languages and software also have differing ways of handling this expression.

**Pull request merged**